app/lib/features/home/widgets/custom_selected_icon.dart

```diff
@@ -1,21 +1,32 @@
+import 'package:acter/features/home/data/keys.dart';
+import 'package:acter/features/news/providers/news_providers.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-class CustomSelectedIcon extends StatelessWidget {
+class CustomSelectedIcon extends ConsumerWidget {
   final Widget icon;
+
   const CustomSelectedIcon({super.key, required this.icon});
 
   @override
-  Widget build(BuildContext context) {
-    return Container(
-      height: 30,
-      width: 54,
-      margin: const EdgeInsets.symmetric(horizontal: 8),
-      padding: const EdgeInsets.only(bottom: 2),
-      decoration: BoxDecoration(
-        color: Theme.of(context).colorScheme.primary,
-        borderRadius: BorderRadius.circular(14),
+  Widget build(BuildContext context, WidgetRef ref) {
+    return InkWell(
+      onDoubleTap: () {
+        if (key == MainNavKeys.updates) {
+          ref.read(boostGoToPageProvider.notifier).state = 0;
```

Comment:
not a fan of having a _specific_ behavior for a single item in the if based on the key in a generic widget. That's not very flutter idiomatic. How about we pass a custom `onDoubleTap` instead?

Reply:
> How about we pass a custom onDoubleTap instead?

hmm.. this look better solution.. Let me change it accordingly.